examples/CMakeLists.txt

```diff
@@ -24,6 +19,15 @@ if(ADIOS2_BUILD_EXAMPLES_EXPERIMENTAL)
   add_subdirectory(experimental)
 endif()
 
-if(ADIOS2_HAVE_CUDA)
+if(ADIOS2_HAVE_CUDA OR ADIOS2_HAVE_Kokkos_CUDA)
   add_subdirectory(cuda)
 endif()
+
+if(ADIOS2_HAVE_Kokkos)
+  add_subdirectory(kokkos)
+else()
+  find_package(Kokkos QUIET)
+  if(Kokkos_FOUND)
+    add_subdirectory(kokkos)
+  endif()
```

Comment:
I am not sure, maybe I am wrong, but perhaps we do not need this code block.

Reply:
We do, this is a Kokkos example that can be used with any backend (it could be CUDA in which case Kokkos is not found before this example)

Reply:
I see, that part of for the case where you build CUDA but you still want to build the Kokkos example code.

But that will enable the kokkos example even if you set `-DADIOS2_US_Kokkos=OFF`

Reply:
Yes `ADIOS2_USE_Kokkos` will be off so we use the CUDA backend underneath but if root to the Kokkos installation will exist the Kokkos will be found. I tried this on Summit and it works.

Reply:
We want to support Kokkos applications even if they only use ADIOS2 with CUDA and not with Kokkos so why not enable the example for both cases?

Reply:
I think the devs and users expect the kokkos example to be disable when they  set `-DADIOS2_USE_Kokkos=OFF` what if I am in a machine with a incompatible kokkos, having this will break my build.

Reply:
Yes I see your point.
It's a tough choice because if I only allow it with the Kokkos backend active it will suggest you can only use Kokkos applications with the Kokkos backend. 
And since there is really no such thing as incompatible Kokkos with the example (it can use any version of Kokkos) I tend to leave it like this. 

I could remove it completely and create a new PR with a Kokkos example so we can debate what's the best solution.

Reply:
I researched it a bit and I am confident we should keep it, the example works with any version of Kokkos, there is nothing that can conflict, and users have a choice of adding -D CMAKE_DISABLE_FIND_PACKAGE_Kokkos=OFF if they want to deactivate it.

Reply:
@anagainaru I am still against this forced Kokkos dependency. In my opinion is unreasonable to expect the user to figure out the `CMAKE_DISABLE_FIND_PACKAGE_Kokkos` option if their build does not work, and while compatible at this day to many kokkos versions there is not guarantee in the future or what if an user has a broken Kokkos installation, he would no be able to bypass this, normally this is not a problem when you can just disable the feature, but here is being forced.

How about this, this is an useful example that uses kokkosView and does not need the Kokkos backend per se, as such that it is not represented by the ADIOS_USE_KOKKOS var. How about adding a new option/var that enables/disable this corner case. something akin to `ADIOS2_USE_EXAMPLE_KOKKOSVIEW`.

Reply:
Ok open an issue for this, this PR is not changing the status quo of the example, what you are describing was a problem before this PR as well so we shouldn't solve it here. But I agree we should discuss it.
I would also be ok moving this example in the `ADIOS2-Examples` repo (maybe as part of an application or benchmark)

Reply:
You are right that was here before this PR. On the other hand we can resolve this by adding in the elseif(ADIOS2_USE_EXAMPLE_KOKKOSVIEW) or similar for instance, in that manner we can avoid the issue, PR, and backport.

Reply:
I would prefer not adding a new option for activating examples in this PR. I think we should discuss the best option.

Reply:
I think we should resolve this in this PR, this PR adds the `ADIOS_USE_KOKKOS` for Kokkos backends, thus it should refactor the code so that it remains consistent. Right now it is not consistent because if ADIOS_USE_KOKKOS=OFF it will still `find_package(Kokkos)` and build the kokkosview example.

The KokkosView example is an exception because it can use CUDA backend and KokkosView (not backend), so a solution would be to be enable it optionally. We also already set by default `ADIOS_USE_KOKKOS=OFF` and `ADIOS_USE_CUDA=OFF` what is the problem with doing the same for this example? We should keep these features optional.

Reply:
@anagainaru we have however been discussing this for a few days, lets move forward and address this later on. I will follow-up with a PR for this.

Reply:
With `ADIOS_USE_CUDA=OFF` we still compile the CUDA example (that will use the Kokkos backend) so in a way this is similar.

We can talk about changing `ADIOS2_USE_CUDA` to `ADIOS2_USE_CUDA_BACKEND` (and same for Kokkos) if you think it's confusing right now but I don't like introducing something just for one example. Let's discuss Tuesday with everyone. For now I will rebase and merge this. Thanks for looking over this.